pkg/controller/templatestack/manager/manager.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manager
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/crossplaneio/crossplane-runtime/pkg/logging"
+	"github.com/crossplaneio/crossplane/apis/stacks/v1alpha1"
+)
+
+const (
+	controllerName   = "templatestackmanager.stacks.crossplane.io"
+	reconcileTimeout = 1 * time.Minute
+	// requeueOnWait    = 30 * time.Second
+)
+
+var log = logging.Logger.WithName("controller." + controllerName)
+
+// A Reconciler reconciles Unstructured objects.
+type Reconciler struct {
+	client client.Client
+	stack  types.NamespacedName
+}
+
+// Controller is responsible for adding the Unstructured
+// controller and its corresponding reconciler to the manager with any runtime configuration.
+type Controller struct {
+	Stack types.NamespacedName
+}
+
+// SetupWithManager creates a new Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func (c *Controller) SetupWithManager(mgr ctrl.Manager) error {
+	client := mgr.GetClient()
+
+	r := &Reconciler{
+		client: client,
+		stack:  c.Stack,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), reconcileTimeout)
+	defer cancel()
+
+	findstack := &unstructured.Unstructured{}
```

Comment:
https://github.com/crossplaneio/crossplane/pull/955/files#diff-92d5ac2f7e08cf29a85448c31cc55436R69-R85
`stack` and `findstack` could have been a single variable, there was no reason to fetch this into an unstructured and marshal/unmarshal later. This was something I was testing that shouldn't have made it into the commit.